zcash_client_sqlite/src/chain.rs

```diff
@@ -1,16 +1,80 @@
 //! Functions for enforcing chain validity and handling chain reorgs.
 use protobuf::Message;
 
-use rusqlite::params;
+use rusqlite::{params};
 
-use zcash_primitives::consensus::BlockHeight;
+use zcash_primitives::{
+    transaction::{components::Amount, TxId},
+    consensus::{self, BlockHeight},
+};
 
-use zcash_client_backend::{data_api::error::Error, proto::compact_formats::CompactBlock};
+use zcash_client_backend::{
+    data_api::error::Error, 
+    proto::compact_formats::CompactBlock,
+    address::RecipientAddress,
+    };
 
 use crate::{error::SqliteClientError, BlockDB};
 
 pub mod init;
 
+
+pub struct UnspentTransactionOutput {
+    pub address: RecipientAddress,
+    pub txid: TxId,
+    pub index: i32, 
+    pub script: Vec<u8>,
+    pub value: Amount,
+    pub height: BlockHeight,
+}
+
+pub fn get_confirmed_utxos_for_address<P: consensus::Parameters>(
+    params: &P,
+    cache: &BlockDB,
+    anchor_height: BlockHeight,
+    address: &str 
+) -> Result<Vec<UnspentTransactionOutput>,SqliteClientError> {
+    let mut stmt_blocks = cache.0.prepare(
```

Comment:
What is inserting data into this table?

Reply:
The rust welding on mobile wallets. The idea is that libruszcash only reads from this table as it only reads the compact_blocks table. Should I add the insertion code as well to make this 'self-sufficient' ? what's the compact_blocks approach?